Port replication_pad1d to ATen (#15485)

lhuang04 opened the pull request:
Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/15485

port replication_pad1d

Differential Revision: D13531920

ezyang commented:
What was the other problem in the end?

ezyang commented on aten/src/ATen/native/cuda/ReplicationPadding.cu:
```diff
@@ -0,0 +1,254 @@
+
+
+      if (numInputDims == 2) {
+      output.resize_({numPlanes, outputW});
```
nit: Indentation here is insufficient

ezyang commented on aten/src/ATen/native/cuda/ReplicationPadding.cu:
```diff
@@ -0,0 +1,254 @@
+      auto gradInput_ = gradInput;
+      auto gradOutput_ = gradOutput;
+      if (numInputDims == 2) {
+      gradInput_ = gradInput.reshape({1, gradInput.size(0),
```
nit: Insufficient indentation here

ezyang reviewed (approved):
I have some blocking comments, but since I'm going to be incommunicado soon, I'm going ahead and accepting this. Please address the comments before landing.

lhuang04 committed:
Port replication_pad1d to ATen (#15507)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/15507

Pull Request resolved: https://github.com/pytorch/pytorch/pull/15485

port replication_pad1d

Reviewed By: ezyang

Differential Revision: D13531920

fbshipit-source-id: f68b01d93c66991e099b7613bdcf380599be5db4
